src/org/thoughtcrime/securesms/util/Util.java

```diff
@@ -344,4 +345,12 @@ public static boolean isClickedRecently() {
     lastClickTime = now;
     return false;
   }
+
+  public static boolean isTouchExplorationEnabled(Context context) {
+    try {
+      return ((AccessibilityManager) context.getSystemService(Context.ACCESSIBILITY_SERVICE)).isTouchExplorationEnabled();
+    } catch (Exception e) {
+      return false;
+    }
+  }
```

Comment:
we're no calling this function not only on a single click, but now also for every message, this easily sums up to some 10 calls per view

as it is unclear how expensive `getSystemSerivce()` actually is, i would cache this call somewhere

there is some discussion at https://stackoverflow.com/questions/32417398/is-context-getsystemservice-an-expensive-call - however, regarding all the different android versions, probably no one can really tell :)

ah, and there is also the hint to go over the applicationContent to be sure ...